ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/PutBulk.java

```diff
@@ -219,4 +219,64 @@ public SeekableByteChannel buildChannel(final String s) throws IOException {
             return this.objectPutter.buildChannel(s);
         }
     }
+
+    private class ObjectsForDirectory {
+
+        private ImmutableList<IgnoreFile> ignoredFiles;
+
+        public ObjectsForDirectory() {
+            //pass
+        }
+
+        public ImmutableList<Ds3Object> getDs3Objects(final Iterable<Path> filteredObjects) throws IOException {
+            final ImmutableList.Builder<Ds3Object> objectsBuilder = ImmutableList.builder();
+            final ImmutableList.Builder<IgnoreFile> ignoredBuilder = ImmutableList.builder();
+
+            for (final Path path : filteredObjects) {
+                try {
+                    objectsBuilder.add(new Ds3Object(
+                            Utils.getFileName(inputDirectory, path),
+                            Utils.getFileSize(path)));
+                } catch (final IOException ex) {
+                    if (!ignoreErrors) throw ex;
+                    LOG.warn(String.format("WARN: file '%s' has an error and will be ignored", path.getFileName()));
+                    ignoredBuilder.add(new IgnoreFile(path, ex.toString()));
+                }
+            }
+
+            ignoredFiles = ignoredBuilder.build();
```

Comment:
Hmmm.  I typically don't encourage methods that have side effects like this.  It seems like this method should be static, and that it returns an object that has both a list of objects to act on, and the list of ignored files.